src/network/ip_address_pool.cpp

```diff
@@ -112,6 +112,24 @@ mp::IPAddress mp::IPAddressPool::obtain_ip_for(const std::string& name)
     return it->second;
 }
 
+mp::IPAddress mp::IPAddressPool::check_ip_for(const std::string& name)
+{
+    const auto it = ip_map.find(name);
+    if (it == ip_map.end())
+    {
+        return mp::IPAddress{""};
```

Comment:
This looks like it should return a std::experimental::optional<IPAddress> instead.

Reply:
That's interesting and didn't know about that:)  But what does that gain us over what I'm already doing here?

Seems I would still need logic to either return the found IPAddress object or {} if not found and the caller would need to use value_or().

Reply:
It would cleanly separate concerns. Instead of trying to make IPAddress a mutable object and have a reserved "invalid" value, you can leave IPAddress as an immutable design and have the "invalid/valid" be represented by the optional type instead - this has all the semantics that you use with the proposed code.